x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/ReservedFieldNames.java

```diff
@@ -175,14 +277,19 @@
      */
     public static boolean isValidFieldName(String fieldName) {
         String[] segments = DOT_PATTERN.split(fieldName);
-        return !RESERVED_FIELD_NAMES.contains(segments[0]);
+        return RESERVED_RESULT_FIELD_NAMES.contains(segments[0]) == false && RESERVED_CONFIG_FIELD_NAMES.contains(segments[0]) == false;
```

Comment:
Is it necessary for the config field names to be included here?  It will prevent fields being added to results documents just because they clash with a config field name.  But that's not necessary because the config and results are in different indices.  Is there another reason?

Reply:
Good catch